qom/object.c

```diff
@@ -1179,10 +1179,10 @@ void object_unref(void *objptr)
     if (!obj) {
         return;
     }
-    g_assert(obj->ref > 0);
+    g_assert(qatomic_mb_read(&obj->ref) > 0);
```

Comment:
object_unref in current QEMU is still the same of this one

```
void object_unref(void *objptr)
{
    Object *obj = OBJECT(objptr);
    if (!obj) {
        return;
    }
    g_assert(obj->ref > 0);

    /* parent always holds a reference to its children */
    if (qatomic_fetch_dec(&obj->ref) == 1) {
        object_finalize(obj);
    }
}
```

If you spotted a bug in QEMU you should report to them

Reply:
`if (qatomic_fetch_dec(&obj->ref) == 1) {` is equal to `if (qatomic_dec_fetch(&obj->ref) == 0) {`, we can left it as is and the patch above would still working. 
Anyway, we are fuzzing and don't care about hw emulation, so we can just change the things.
Same method, as with `rcu_disable_atfork()`, right?

`/* parent always holds a reference to its children */`
so the child left allocated forever.